bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigUtil.java

```diff
@@ -161,4 +128,175 @@ public static Object normalizeType(Object value, ConfigDescriptionParameter conf
         return convertedConfiguration;
     }
 
+    /**
+     * Normalizes the type of the parameter to the one allowed for configurations.
+     *
+     * @param value the value to return as normalized type
+     * @return corresponding value as a valid type
+     * @deprecated Use {@link #normalizeType(Object, ConfigDescriptionParameter)} instead
+     */
+    @Deprecated
+    public static Object normalizeType(Object value) {
+        return normalizeType(value, null);
+    }
+
+    /**
+     * Use this method to automatically map a configuration collection to a Configuration holder object. A common
+     * use-case would be within a service. Usage example:
+     *
+     * <pre>
+     * {@code
+     *   public void modified(Map<String, Object> properties) {
+     *     YourConfig config = ConfigUtil.as(properties, YourConfig.class);
+     *   }
+     * }
+     * </pre>
+     *
+     *
+     * @param properties The configuration map.
+     * @param configurationClass The configuration holder class. An instance of this will be created so make sure that
+     *            a default constructor is available.
+     * @return The configuration holder object. All fields that matched a configuration option are set. If a required
+     *         field is not set, null is returned.
+     */
+    public static <T> @Nullable T as(Map<String, Object> properties, Class<T> configurationClass) {
```

Comment:
Do you have any particular reasons why this method must/should be exposed as API? I'd rather have expected to see in in a class within an internal package. In that way the `Configuration` class can use it and everybody else can do a `new Configuration(Map).as(Class)` if needed. What did I miss?

Reply:
@sjka I tried to explain this in the PR message :) 
`new Configuration(Map).as(Class)` does more than we need for service configuration objects. It "normalizes" values, which basically means all numbers -> BigDecimal, Collections -> Lists.
But the ConfigUtils.as method is way smarter, because it converts the types based on the field type. So yes, it makes sense to have it public API and use ConfigUtils.as within services.

Reply:
> I tried to explain this in the PR message

If I didn't get that from your explanation, how should a user of the API figure that out? That is pretty surprising. 

Another thing worth mentioning is that static utility methods only got us into trouble, both on testing/mocking side as well as with respect to evolving the APIs. We have been working in the past on getting rid of several such static helper methods and that always was painful on some end. Therefore I'd like to object to adding yet another one.

> Maybe it should be marked as deprecated so that nobody starts to use it before this is discussed.

I don't know which "as" method you mean: 
* Configuration.as(...) must stay where it is, that cannot be moved anymore.
* ConfigUtil.as.(...) must not be introduced publicly if we already suspect it will have to move. Why does it have to be public then if nobody should start using it?